Add support for ReSignIn

HaoK opened the pull request:
- Add ResignIn api to regenerate the application cookie preserving rememberMe and AuthenticationMethod
- Update manage controller in samples to use new API

Fixes: https://github.com/aspnet/Identity/issues/185

@divega @Tratcher @suhasj @rustd @blowdart

suhasj commented on src/Microsoft.AspNet.Identity/SignInManager.cs:
```diff
@@ -75,6 +75,21 @@ public class SignInManager<TUser> where TUser : class
+        {
+            var authResult = await Context.AuthenticateAsync(IdentityOptions.ApplicationCookieAuthenticationScheme);
+            var properties = authResult?.Properties;
+            var isPersistent = properties != null && properties.IsPersistent;
```
So we are preserving only the `IsPersistent` property ? The other property I feel might be slightly important is the Dictionary. Do you think we need that too ?

HaoK replied:
@Tratcher what do you think we should preserve when it comes to the AuthenticationProperties, we definitely want to regenerate the claims, but we want to keep as many of the old settings as possible that make sense...

Tratcher replied:
It's a generic property bag, you'd want to preserve anything you can.

HaoK replied:
Is it safe to copy everything over then?  Except for maybe nulling out ExpiresUtc and IssuedUtc unless SignIn will take care of restamping those for us...

HaoK replied:
@divega We can consider changing SignIn to mimic the underlying SignIn API more closely, remove IsPersistent and take in a raw AuthenticationProperties which would expose everything

HaoK commented:
Updated, kept old SignIn overload which just takes IsPersistence, but added an overload which takes AuthenticationProperties, and switched ReSignInAsync to use that.

divega commented:
I am ok with this, but didn't you say that we were the main users of this API and it didn't make sense to keep the overload that takes a bool? Anything changed your point of view?

HaoK commented:
Well, I tried it without the overload, and it was annoying and i immediately wrote a helper overload, so might as well keep it and leave it public (no breaking change at all this way too).  So that is what convinced me to keep it :)

divega commented:
Fair enough :smile: :shipit:

rustd commented:
The name of the method is not great. How about RefreshIdentity? RegenerateIdentity?

divega commented:
Fair point too. Seems to me that the word "renew" may be closer in meaning to what the API will do.

divega commented:
E.g. 'RenewSignIn' or something like that.

HaoK commented:
RenewSignIn or RefreshSignIn works for me

rustd commented:
Either one is good. Let's pick RefreshSignIn?
On Thu, Apr 23, 2015 at 7:58 AM Hao Kung <notifications@github.com> wrote:

> RenewSignIn or RefreshSignIn works for me
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/aspnet/Identity/pull/452#issuecomment-95614420>.
>

HaoK commented:
Ok updated to RefreshSignIn

divega commented:
:ship it:

HaoK commented:
c251bf0162091730b6cc74d97731db74d98e499d
